src/test/java/com/staxrt/tutorial/ApplicationTests.java

```diff
@@ -0,0 +1,93 @@
+package com.staxrt.tutorial;
+
+import com.staxrt.tutorial.model.User;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.HttpClientErrorException;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Application.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ApplicationTests {
```

Comment:
1. Zrób testy dla obsługi wpisów a nie dla aplikacji. Szybko ta klasa za mocno spuchnie.
2. W tej chwili testy używają tego samego application.properties co kod produkcyjny. To zmusza do tego, że developer przeglądający zmianę, żeby sprawdzić czy to w ogóle działa, musi zainstalować w swoim systemie mysql, a to jest niedopuszczalne. Najlepiej by było jakbyś użył: https://www.testcontainers.org/ i w testach stawaiał mysql na dockerze. Alternatywnie możesz użyć w testach jakiejś bazy pamięciowej np. https://www.h2database.com/html/main.html. Ostatecznie możesz napisać prostego docker-compose.yml z mysql i spisać w instrukcji co trzeba zrobić, żeby wystartować testy tzn. `docker-compose up` a potem `mvn test`.

Reply:
Chciałem w wersji V2 wrzucić dockera bo w planach mam robić to kompatybilne z dockerem ale jeśli to jest niedopuszczalne to dla wersji v1 mogę już dockera wrzucić

Reply:
fajnie by było bo z chęcią bym sprawdził czy testy zadziałają, natomiast może to być na razie bardzo proste podejście, bez testcointainers, tylko baza danych w dockerze. tak żeby reviewer musiał wykonać jak najmniej kroków, żeby odpalić testy. w tej chwili krzyczy mi że nie mam mysql a nie chcę go instalować

Reply:
Well tutaj mam komplikacje poniewaz dodalem oddzielny properties dla testow, przerobilem poma calkowicie od 0 za pomoca spring initializer ale z jakis przyczyn przestal mi calkowicie robic testy jak wykonuje mvn test. wykonuje 0 testow i przechodzi pozytywnie. Za to jak sie odpali recznie testy startujac klase applicationtest to wszystko dziala poprawnie

Reply:
poprawnie tzn mi nie dziala te h2, wyskakuje mi ze Cannot load driver class: org.h2.Driver

Reply:
ale jak properties w testach przeksztalcam na mysql to wszystko smiga tak wiec odzielny properties dla testow dziala, co do testcointainters bedziesz moze mogl mi dac jakies wskazowki albo pokazac to jak skonfigurowac badz o co chodzi z tym h2. Testy wyrzucilem zostal sam test contextload()

Reply:
Odnośnie tego, to pamiętaj, żeby po zmianie poma reimportować projekt w idei. Puściłem z command line `mvn clean test` i brakuje zależności do `junit`. Przy okazji przejdź tę instrukcję i zamiast junit4 użyj junit5: https://dev.to/martinbelev/how-to-enable-junit-5-in-new-spring-boot-project-29a8 . Po tych krokach u mnie test zaczął się odpalać z cmd, tylko krzyczy `Table "HIBERNATE_SEQUENCE" not found`, ale to dlatego że używasz dialectu mysql zamiast h2. Po zmianie dialectu śmiga jak należy.

Reply:
done